server/src/main/java/org/elasticsearch/search/vectors/KnnSearchRequestParser.java

```diff
@@ -194,6 +196,7 @@ public void toSearchRequest(SearchRequestBuilder builder) {
 
     // visible for testing
     static class KnnSearch {
```

Comment:
We should not update this class anylonger. Its deprecated and folks shouldn't use the `_knn_search` API.

Reply:
Rolled back all changes to `KnnSearch`  in 1aa708b41edcc59ee63c4f10c61978c9b7147a75